packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/PropertyProvider.cs

```diff
@@ -17,6 +17,7 @@ public class PropertyProvider
         public XmlDocSummaryStatement XmlDocSummary { get; }
         public MethodSignatureModifiers Modifiers { get; }
         public CSharpType Type { get; }
+        public bool IsRequired { get; }
```

Comment:
This property only applies during serialization.  For dotnet properties there is no concept of "requiredness".

I wonder if a PropertyProvider should have a SerializationInfo structure on it which contains things like
- IsRequired
- SerializationFormat
- etc